tests/integration_tests/test_pruned_node.py

```diff
@@ -0,0 +1,120 @@
+from pathlib import Path
+
+import pytest
+from eth_bloom import BloomFilter
+from eth_utils import abi, big_endian_to_int
+from hexbytes import HexBytes
+from web3.datastructures import AttributeDict
+
+from .network import setup_custom_ethermint
+from .utils import (
+    ADDRS,
+    CONTRACTS_PATHS,
+    KEYS,
+    deploy_contract,
+    sign_transaction,
+    wait_for_new_blocks,
+)
+
+
+@pytest.fixture(scope="module")
+def ethermint(request, tmp_path_factory):
+    """start-ethermint
+    params: enable_auto_deployment
+    """
+    yield from setup_custom_ethermint(
+        tmp_path_factory.mktemp("pruned"),
+        26900,
+        Path(__file__).parent / "configs/pruned-node.jsonnet",
+    )
+
+
+def test_pruned_node(ethermint):
+    """
+    test basic json-rpc apis works in pruned node
+    """
+    w3 = ethermint.w3
+    erc20 = deploy_contract(
+        w3,
+        CONTRACTS_PATHS["TestERC20"],
+        key=KEYS["validator"],
+    )
+    tx = erc20.functions.transfer(ADDRS["community"], 10).buildTransaction(
+        {"from": ADDRS["validator"]}
+    )
+    signed = sign_transaction(w3, tx, KEYS["validator"])
+    txhash = w3.eth.send_raw_transaction(signed.rawTransaction)
+
+    print("wait for prunning happens")
+    wait_for_new_blocks(ethermint.cosmos_cli(0), 10)
+
+    txreceipt = w3.eth.wait_for_transaction_receipt(txhash)
+    assert len(txreceipt.logs) == 1
+    expect_log = {
+        "address": erc20.address,
+        "topics": [
+            HexBytes(
+                abi.event_signature_to_log_topic(
+                    "Transfer(address,address,uint256)")
+            ),
+            HexBytes(b"\x00" * 12 + HexBytes(ADDRS["validator"])),
+            HexBytes(b"\x00" * 12 + HexBytes(ADDRS["community"])),
+        ],
+        "data": "0x000000000000000000000000000000000000000000000000000000000000000a",
+        "transactionIndex": 0,
+        "logIndex": 0,
+        "removed": False,
+    }
+    assert expect_log.items() <= txreceipt.logs[0].items()
+
+    # check get_balance and eth_call don't work on pruned state
+    # we need to check error message here. 
+    # `get_balance` returns unmarshallJson and thats not what it should
```

Comment:
The json-rpc error message is not friendly in this case, maybe sth to improve later.